### PR TITLE
fix(Core/Spell): Make use of SPELL_ATTR7_IGNORE_COLD_WEATHER_FLYING

### PR DESCRIPTION
**Pull request by Kitzunu**
## Changes Proposed:
-  Make use of SPELL_ATTR7_IGNORE_COLD_WEATHER_FLYING
-  Affected spells:

```sql
SELECT `id`, `name_lang_1`, `attributesEx7`
FROM `dbc_spell` 
WHERE `attributesEx7` & @flag = @flag
```
| id | name_lang_1 | attributesEx7 | 
| ---: | --- | ---: | 
| 64681 | Loaned Gryphon | 4194304 | 
| 64749 | Loaned Gryphon | 4194304 | 
| 64761 | Loaned Wind Rider | 4194304 | 
| 64762 | Loaned Wind Rider | 4194304 | 
| 72422 | Quivering Strike | 4194304 | 
| 72453 | Quivering Strike | 4194304 | 

## Tests Performed:
- Built
- Tested in-game

## How to Test the Changes:

1. go on a char that does not have cold weather flying
2. .learn 64681
3. See you can mount that
4. .learn 32240
5. See you cannot mount that
6. .learn 54197
7. See you can mount both

## Target Branch(es):
- [x] Master

 
## How to Test AzerothCore PRs
 
When a PR is ready to be tested, it will be marked as **[WAITING TO BE TESTED]**.

You can help by testing PRs and writing your feedback here in the PR's page on GitHub. Follow the instructions here:

http://www.azerothcore.org/wiki/How-to-test-a-PR

### CODE REVIEW

**Comment by Kitzunu** [comment]
https://github.com/azerothcore/azerothcore-wotlk/issues/4306

**Comment by pklloveyou** [comment]
SpellInfo.cpp
if (player && !player->canFlyInZone(map_id, zone_id))

**Comment by Kitzunu** [comment]
Done

File: src/server/shared/DataStores/DBCStructure.h [hunk 1]
```diff
@@ -522,6 +522,17 @@ struct AreaTableEntry
...
+
+    [[nodiscard]] bool IsFlyable() const
+    {
+        if (flags & AREA_FLAG_OUTLAND)
+        {
+            if (!(flags & AREA_FLAG_NO_FLY_ZONE))
+                return true;
+        }
+
+        return false;
```
Comment (Helias) [inline src/server/shared/DataStores/DBCStructure.h:534]:
```suggestion
        return flags & AREA_FLAG_OUTLAND && !(flags & AREA_FLAG_NO_FLY_ZONE);
```

**Review by Helias — CHANGES_REQUESTED** [review]

File: src/server/game/Spells/SpellInfo.cpp [hunk 2]
```diff
@@ -1473,16 +1473,43 @@ SpellCastResult SpellInfo::CheckLocation(uint32 map_id, uint32 zone_id, uint32 a
...
+        {
+            AreaTableEntry const* areaEntry = sAreaTableStore.LookupEntry(area_id);
+            if (!areaEntry)
+            {
+                areaEntry = sAreaTableStore.LookupEntry(zone_id);
+            }
+
+            if (!areaEntry || !areaEntry->IsFlyable() || !player->canFlyInZone(map_id, zone_id, this))
+            {
+                return SPELL_FAILED_INCORRECT_AREA;
```
Comment (Helias) [inline src/server/game/Spells/SpellInfo.cpp:1497]: you could remove the condition `!areaEntry` and add the return even in the previous if conditions, but it is not so relevant, so your choice
(you only save one double check)

Comment (Kitzunu) [reply src/server/game/Spells/SpellInfo.cpp:1497]: Null check

**Comment by pklloveyou** [comment]
Tested

File: src/server/shared/DataStores/DBCStructure.h (hunk 1 above)
Comment (FrancescoBorzi) [reply src/server/shared/DataStores/DBCStructure.h:534]: @Kitzunu what about this

Comment (Kitzunu) [reply src/server/shared/DataStores/DBCStructure.h:534]: If you prefer this, then I can check it

**Review by FrancescoBorzi — APPROVED** [review]

**Pull request merged**